### ISSUE TITLE
[Python][CI] Tests involving fastparquet are never run

### ISSUE THREAD

**Issue by jorisvandenbossche** [apache/arrow#37853]
We have a `fastparquet` pytest marker for tests that require fastparquet, and we have two such tests, but from a search in our code base, it doesn't seem that we include fastparquet anywhere in one of our CI test builds.

The two tests are:

- `test_fastparquet_cross_compatibility` in the parquet tests (I know that pandas has similar cross compat tests)
- `test_fastparquet_read_with_hdfs` in test_hdfs.py: ensuring fastparquet can use our HDFS filesystem -> this is something that fsspec / fastparquet can test themselves I think, and this is also in our legacy HDFS tests that we will remove once removing the legacy HDFS bindings 

Given the above, we could also consider removing the tests alltogether (although adding fastparquet in one of the nightly builds should also be easy)

**Comment by jorisvandenbossche** [comment]
Context: the parquet cross-compat test appears to be failing when enabled in the conda recipe tests: https://github.com/apache/arrow/pull/37624#discussion_r1328153969

And I can actually reproduce this locally, both with running our test, as with the following small snippet:

```
In [1]: df = pd.DataFrame({"col": [True, False, True]})

In [2]: df.to_parquet("test_bool.parquet", engine="pyarrow")

In [3]: pd.read_parquet("test_bool.parquet", engine="pyarrow")
Out[3]: 
     col
0   True
1  False
2   True

In [4]: pd.read_parquet("test_bool.parquet", engine="fastparquet")
Out[4]: 
     col
0  False
1  False
2  False

In [5]: pa.__version__
Out[5]: '14.0.0.dev183+g7b14b2b27.d20230925'

In [6]: import fastparquet; fastparquet.__version__
Out[6]: '2023.8.0'
```

**Comment by jorisvandenbossche** [comment]
Yes, so it's the different encoding that makes fastparquet fail:

```
In [5]: df = pd.DataFrame({"col": [True, False, True]})

In [8]: df.to_parquet("test_bool_pa14_plain.parquet", engine="pyarrow", column_encoding={"col": "PLAIN"}, use_dictionary=False)

In [9]: df.to_parquet("test_bool_pa14_rle.parquet", engine="pyarrow", column_encoding={"col": "RLE"}, use_dictionary=False)

In [10]: pd.read_parquet("test_bool_pa14_plain.parquet", engine="fastparquet")
Out[10]: 
     col
0   True
1  False
2   True

In [11]: pd.read_parquet("test_bool_pa14_rle.parquet", engine="fastparquet")
Out[11]: 
     col
0  False
1  False
2  False
```

This is getting a bit off-topic for this issue, but maybe that's a good argument to actually _do_ run those tests on our CI, then we would have noticed this compat issue earlier.

**Comment by pitrou** [comment]
Do we still care about this or should we close this issue?

**Comment by pitrou** [comment]
@raulcd We should perhaps simply remove those tests.

**Comment by raulcd** [comment]
I agree, those tests  are also failing and deprecated as fastparquet does not support pandas 3, see related issue:
- https://github.com/apache/arrow/issues/49839

**Comment by raulcd** [comment]
Issue resolved by pull request 50416
https://github.com/apache/arrow/pull/50416
